src/Files.App/ViewModels/ShellViewModel.cs

```diff
@@ -1813,7 +1813,7 @@ private void GetDesktopIniFileData()
 
 		public void CheckForBackgroundImage()
 		{
-			var filesAppSection = DesktopIni.FirstOrDefault(x => x.SectionName == "FilesApp");
+			var filesAppSection = DesktopIni?.FirstOrDefault(x => x.SectionName == "FilesApp");
```

Comment:
@0x5bfa I found that `LayoutModeChangeRequested()` might call this method before desktop.ini was loaded, so I added a null-safe operator to ignore the call.

Reply:
Should we add the loader inside this method? This doesn't make much sense since even if ini file  exists you'll have to reload until it's loaded

Reply:
EnumerateItemsFromStandardFolderAsync should always load desktop.ini when the folder is successfully loaded, so I don't think it is necessary to load it every time.

Reply:
Correct me if I'm wrong

once this desktop background check method called ahead of time, the background is still null. So unless you don't refresh and pray that desktop ini data is loaded first, the background won't be loaded.

Reply:
CheckForBackgroundImage is called without loaded desktop.ini if LayoutModeChangeRequested is called before the folder load is completed or after the folder load fails. But the former will always call GetDesktopIniFileData and CheckForBackgroundImage in EnumerateItemsFromStandardFolderAsync, and the latter is an error state and can be ignored.

Reply:
Dang. Should we set a setter to call CheckForBackgroundImage in the property?

Reply:
That's one way. I don't have a preference.

Reply:
If the Folder background has a fade in transition, then you can load the folder view, and once the contents are done loading, you can then check and fade in the folder image, if there is an ini file.  Lazy load essentially.